sympy/ntheory/generate.py

```diff
@@ -745,14 +747,37 @@ def primerange(a, b=None):
         a, b = 2, a
     if a >= b:
         return
-    # if we already have the range, return it
-    if b <= sieve._list[-1]:
+    # If we already have the range, return it.
+    largest_known_prime = sieve._list[-1]
+    if b <= largest_known_prime:
         yield from sieve.primerange(a, b)
         return
+    # If we know some of it, return it.
+    if a <= largest_known_prime:
+        yield from sieve.primerange(a, largest_known_prime + 1)
+        a = largest_known_prime + 1
+    if a % 2:
```

Comment:
Is this in case the largest_known_prime is 2?

Reply:
Implicitly assumes `2 < largest_known_prime`. This is because `_list[-1]` is never less than 13 in the `Sieve` constructor and `_reset`.

Reply:
Looking at it today, I think I was confusing `a` with something on the sieve; but `a` is just a given bound (which could be odd or even).